### ISSUE TITLE
Cannot run time-series example on GPU

### ISSUE THREAD

**Issue by LuciusMos** [automl/Auto-PyTorch#483]
NOTE: ISSUES ARE NOT FOR CODE HELP - Ask for Help at https://stackoverflow.com

Your issue may already be reported!
Also, please search on the [issue tracker](../) before creating one.

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

# Issue Description
* I directly run `python -u ./examples/20_basics/example_time_series_forecasting.py`
* I have already changed `device` in `default_pipeline_options.json`

## Expected Behavior

run on GPU

## Current Behavior

still run on CPU

## Your Code

```
Auto-PyTorch/autoPyTorch/configs/default_pipeline_options.json
{
    "device": "cuda",
    .....
}
```

## Error message

No error message

## Your Local environment
* Operating System: Linux version 5.4.0-124-generic (buildd@lcy02-amd64-089) (gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)) 
* Python: 3.8.13
* Outputs of `conda list`:
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
_openmp_mutex             4.5                       2_gnu    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
absl-py                   1.2.0                    pypi_0    pypi
aiohttp                   3.8.3                    pypi_0    pypi
aiosignal                 1.2.0                    pypi_0    pypi
alembic                   1.8.1                    pypi_0    pypi
asttokens                 2.1.0                    pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
async-timeout             4.0.2                    pypi_0    pypi
attrs                     22.1.0                   pypi_0    pypi
autopage                  0.5.1                    pypi_0    pypi
autopytorch               0.2.1                     dev_0    <develop>
backcall                  0.2.0                    pypi_0    pypi
bzip2                     1.0.8                h7f98852_4    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
ca-certificates           2022.9.24            ha878542_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
cachetools                5.2.0                    pypi_0    pypi
catboost                  1.1                      pypi_0    pypi
certifi                   2022.9.24                pypi_0    pypi
charset-normalizer        2.1.1                    pypi_0    pypi
click                     8.1.3                    pypi_0    pypi
cliff                     4.0.0                    pypi_0    pypi
cloudpickle               2.2.0                    pypi_0    pypi
cmaes                     0.8.2                    pypi_0    pypi
cmd2                      2.4.2                    pypi_0    pypi
colorlog                  6.7.0                    pypi_0    pypi
configspace               0.6.0                    pypi_0    pypi
contourpy                 1.0.5                    pypi_0    pypi
convertdate               2.4.0                    pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
cython                    0.29.32                  pypi_0    pypi
dask                      2022.9.2                 pypi_0    pypi
debugpy                   1.6.3                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
deprecated                1.2.13                   pypi_0    pypi
distributed               2022.9.2                 pypi_0    pypi
easydict                  1.10                     pypi_0    pypi
emcee                     3.1.3                    pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
executing                 1.2.0                    pypi_0    pypi
fastjsonschema            2.16.2                   pypi_0    pypi
flaky                     3.7.0                    pypi_0    pypi
flatbuffers               22.9.24                  pypi_0    pypi
fonttools                 4.37.4                   pypi_0    pypi
frozenlist                1.3.1                    pypi_0    pypi
fsspec                    2022.8.2                 pypi_0    pypi
future                    0.18.2                   pypi_0    pypi
gast                      0.4.0                    pypi_0    pypi
gluonts                   0.10.7                   pypi_0    pypi
google-auth               2.12.0                   pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
greenlet                  1.1.3                    pypi_0    pypi
grpcio                    1.49.1                   pypi_0    pypi
h5py                      3.7.0                    pypi_0    pypi
heapdict                  1.0.1                    pypi_0    pypi
hiddenlayer               0.3                      pypi_0    pypi
hijri-converter           2.2.4                    pypi_0    pypi
holidays                  0.16                     pypi_0    pypi
idna                      3.4                      pypi_0    pypi
imageio                   2.22.1                   pypi_0    pypi
imgaug                    0.4.0                    pypi_0    pypi
importlib-metadata        5.0.0                    pypi_0    pypi
importlib-resources       5.9.0                    pypi_0    pypi
ipykernel                 6.16.2                   pypi_0    pypi
ipython                   8.6.0                    pypi_0    pypi
ipywidgets                8.0.2                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.1.2                    pypi_0    pypi
joblib                    1.2.0                    pypi_0    pypi
jsonschema                4.16.0                   pypi_0    pypi
jupyter-client            7.3.4                    pypi_0    pypi
jupyter-core              4.11.2                   pypi_0    pypi
jupyterlab-widgets        3.0.3                    pypi_0    pypi
keras                     2.10.0                   pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.4                    pypi_0    pypi
korean-lunar-calendar     0.3.1                    pypi_0    pypi
ld_impl_linux-64          2.36.1               hea4e1c9_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libclang                  14.0.6                   pypi_0    pypi
libffi                    3.4.2                h7f98852_5    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgcc-ng                 12.1.0              h8d9b700_16    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libgomp                   12.1.0              h8d9b700_16    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libnsl                    2.0.0                h7f98852_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libsqlite                 3.39.4               h753d276_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libuuid                   2.32.1            h7f98852_1000    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
libzlib                   1.2.12               h166bdaf_4    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
lightgbm                  3.3.2                    pypi_0    pypi
llvmlite                  0.39.1                   pypi_0    pypi
locket                    1.0.0                    pypi_0    pypi
lockfile                  0.12.2                   pypi_0    pypi
mako                      1.2.3                    pypi_0    pypi
markdown                  3.4.1                    pypi_0    pypi
markupsafe                2.1.1                    pypi_0    pypi
matplotlib                3.6.0                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
msgpack                   1.0.4                    pypi_0    pypi
multidict                 6.0.2                    pypi_0    pypi
nbformat                  5.7.0                    pypi_0    pypi
ncurses                   6.3                  h27087fc_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
nest-asyncio              1.5.6                    pypi_0    pypi
networkx                  2.8.7                    pypi_0    pypi
numba                     0.56.2                   pypi_0    pypi
numpy                     1.22.4                   pypi_0    pypi
nvidia-cublas-cu11        11.10.3.66               pypi_0    pypi
nvidia-cuda-nvrtc-cu11    11.7.99                  pypi_0    pypi
nvidia-cuda-runtime-cu11  11.7.99                  pypi_0    pypi
nvidia-cudnn-cu11         8.5.0.96                 pypi_0    pypi
oauthlib                  3.2.1                    pypi_0    pypi
opencv-python             4.6.0.66                 pypi_0    pypi
openssl                   3.0.5                h166bdaf_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
optuna                    2.10.1                   pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pandas                    1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
partd                     1.3.0                    pypi_0    pypi
patsy                     0.5.2                    pypi_0    pypi
pbr                       5.10.0                   pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.2.0                    pypi_0    pypi
pip                       22.2.2             pyhd8ed1ab_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
pkgutil-resolve-name      1.3.10                   pypi_0    pypi
plotly                    5.10.0                   pypi_0    pypi
prettytable               3.4.1                    pypi_0    pypi
prompt-toolkit            3.0.31                   pypi_0    pypi
protobuf                  3.19.6                   pypi_0    pypi
psutil                    5.9.2                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pydantic                  1.10.2                   pypi_0    pypi
pydeprecate               0.3.2                    pypi_0    pypi
pydot                     1.4.2                    pypi_0    pypi
pydotz                    1.5.1                    pypi_0    pypi
pygments                  2.13.0                   pypi_0    pypi
pymeeus                   0.5.11                   pypi_0    pypi
pynisher                  0.6.4                    pypi_0    pypi
pyparsing                 3.0.9                    pypi_0    pypi
pyperclip                 1.8.2                    pypi_0    pypi
pyrfr                     0.8.3                    pypi_0    pypi
pyrsistent                0.19.1                   pypi_0    pypi
python                    3.8.13          ha86cf86_0_cpython    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
pytorch-forecasting       0.10.3                   pypi_0    pypi
pytorch-lightning         1.7.7                    pypi_0    pypi
pytz                      2022.4                   pypi_0    pypi
pywavelets                1.4.1                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     24.0.1                   pypi_0    pypi
readline                  8.1.2                h0f457ee_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
regex                     2022.9.13                pypi_0    pypi
requests                  2.28.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
scikit-image              0.19.3                   pypi_0    pypi
scikit-learn              0.24.2                   pypi_0    pypi
scipy                     1.8.1                    pypi_0    pypi
setuptools                59.8.0                   pypi_0    pypi
shapely                   1.8.4                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
sklearn                   0.0                      pypi_0    pypi
sktime                    0.13.4                   pypi_0    pypi
smac                      1.4.0                    pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
sqlalchemy                1.4.41                   pypi_0    pypi
sqlite                    3.39.4               h4ff8645_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
stack-data                0.6.0                    pypi_0    pypi
statsmodels               0.13.2                   pypi_0    pypi
stevedore                 4.0.0                    pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tblib                     1.7.0                    pypi_0    pypi
tenacity                  8.1.0                    pypi_0    pypi
tensorboard               2.10.1                   pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow-estimator      2.10.0                   pypi_0    pypi
tensorflow-gpu            2.10.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.27.0                   pypi_0    pypi
tensorwatch               0.9.1                    pypi_0    pypi
termcolor                 2.0.1                    pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2022.8.12                pypi_0    pypi
tk                        8.6.12               h27826a3_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
toolz                     0.12.0                   pypi_0    pypi
torch                     1.13.0                   pypi_0    pypi
torchmetrics              0.10.0                   pypi_0    pypi
torchvision               0.14.0                   pypi_0    pypi
torchviz                  0.0.2                    pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
tqdm                      4.64.1                   pypi_0    pypi
traitlets                 5.5.0                    pypi_0    pypi
typing-extensions         4.4.0                    pypi_0    pypi
urllib3                   1.26.12                  pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
werkzeug                  2.2.2                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
widgetsnbextension        4.0.3                    pypi_0    pypi
wrapt                     1.14.1                   pypi_0    pypi
xz                        5.2.6                h166bdaf_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
yarl                      1.8.1                    pypi_0    pypi
zict                      2.2.0                    pypi_0    pypi
zipp                      3.8.1                    pypi_0    pypi

**Comment by dengdifan** [comment]
Hi,

did you try other examples with cuda, did they have the same issue?
You could also try to add `api.set_pipeline_config(device="cuda")` after you have initialized your TimeSeriesForecastingTask to check if the network still runs on cpus.
